### PR TITLE
fix: remove rti from navigations

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. removed RTI from `rh-subnav`,`rh-navigation-secondary`, `uxdot-sidenav` as previously discussed with @hellogreg and @adamjohnson 
2. Corrects `rh-subnav` icons.
3. added `accessible-label` to `rh-subnav` and `rh-navigation-secondary`.  

## Testing Instructions

1.

### CODE REVIEW

**Comment by zeroedin** [comment]
considering moving the docs `uxdot-sidenav` fix to its own PR off main to get it out to production faster.

**Comment by bennypowers** [comment]
Should we remove from accordion as well @adamjohnson ?

**Comment by adamjohnson** [comment]
> considering moving the docs `uxdot-sidenav` fix to its own PR off main to get it out to production faster.

Yes, let's get the docs `uxdot-sidenav` fix in a separate PR.

File: elements/rh-navigation-secondary/rh-navigation-secondary.ts [hunk 1]
```diff
@@ -114,6 +103,7 @@ export class RhNavigationSecondary extends LitElement {
 
   @queryAssignedElements({ slot: 'nav' }) private _nav?: HTMLElement[];
 
+  @property({ attribute: 'accessible-label' }) accessibleLabel = 'secondary';
```
Comment (adamjohnson) [inline elements/rh-navigation-secondary/rh-navigation-secondary.ts:106]: Needs a comment above this line to document what this property does.

File: elements/rh-subnav/rh-subnav.ts [hunk 2]
```diff
@@ -63,20 +64,16 @@ export class RhSubnav extends LitElement {
   @colorContextProvider()
   @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette;
 
+  @property({ attribute: 'accessible-label' }) accessibleLabel = 'subnavigation';
```
Comment (adamjohnson) [inline elements/rh-subnav/rh-subnav.ts:67]: Needs a comment above this line to document what this property is for.

File: elements/rh-subnav/rh-subnav.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { LitElement, html } from 'lit';
 import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
```
Comment (adamjohnson) [inline elements/rh-subnav/rh-subnav.ts:3]: `ifDefined` is called but never used. Nuke it.

File: docs/assets/javascript/elements/uxdot-sidenav.ts [hunk 4]
```diff
@@ -4,8 +4,6 @@ import { classMap } from 'lit/directives/class-map.js';
 import { customElement } from 'lit/decorators/custom-element.js';
 import { property } from 'lit/decorators/property.js';
 
-import { RovingTabindexController } from '@patternfly/pfe-core/controllers/roving-tabindex-controller.js';
```
Comment (adamjohnson) [inline docs/assets/javascript/elements/uxdot-sidenav.ts:7]: Move these changes into their own PR to get them into production faster.

File: elements/rh-navigation-secondary/rh-navigation-secondary.ts [hunk 5]
```diff
@@ -6,7 +6,6 @@ import { state } from 'lit/decorators/state.js';
 import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
 
 import { ComposedEvent } from '@patternfly/pfe-core';
-import { RovingTabindexController } from '@patternfly/pfe-core/controllers/roving-tabindex-controller.js';
```
Comment (adamjohnson) [inline elements/rh-navigation-secondary/rh-navigation-secondary.ts:9]: Focus is lost briefly when tabbing, here's how to reproduce:

1. Visit the [DP demo for secondary nav](https://deploy-preview-1821--red-hat-design-system.netlify.app/elements/navigation-secondary/demo/).
2. Hit "Tab" on your keyboard until you get to the first dropdown.
3. Open the first dropdown.
4. Keep hitting Tab to tab through every element in the dropdown.
5. When you get to the last focusable item in the dropdown, hit tab again.
6. **Notice the dropdown disappears but the current focused item is lost.**
7. If you hit Tab one more time, you'll notice focus returns again and is on the next dropdown menu toggle.

Expected results: focus should remain visible for each tab stop.

Related to [WCAG 2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)

If you test this on ux.redhat.com, note the difference in functionality there vs here.

Comment (adamjohnson) [reply elements/rh-navigation-secondary/rh-navigation-secondary.ts:9]: If you test this with VO + Safari, it looks like Safari returns focus to the first focusable element on the page (the logo, in this case).

File: elements/rh-subnav/rh-subnav.ts [hunk 6]
```diff
@@ -94,29 +91,22 @@ export class RhSubnav extends LitElement {
...
-    this.#mo.disconnect();
   }
 
   render() {
     const { scrollIconSet, scrollIconLeft, scrollIconRight } = this.constructor as typeof RhSubnav;
     const { showScrollButtons } = this.#overflow;
     const { on = '' } = this;
     return html`
-      <nav part="container" class="${classMap({ [on]: !!on })}">${!showScrollButtons ? '' : html`
+      <nav part="container" aria-label="${this.accessibleLabel}" class="${classMap({ [on]: !!on })}">${!showScrollButtons ? '' : html`
```
Comment (adamjohnson) [inline elements/rh-subnav/rh-subnav.ts:109]: Don't output the `aria-label` attribute if `accessibleLabel` is empty.

**Review by adamjohnson — CHANGES_REQUESTED** [review]
Some accessibility changes to sort out.

Overall, I'm a big fan of removing RTI where appropriate. I think this will be less confusing to users.

File: elements/rh-subnav/rh-subnav.ts (hunk 6 above)
Comment (zeroedin) [reply elements/rh-subnav/rh-subnav.ts:109]: `accessibleLabel` as coded by default is set to `subnavigation` so it should never be empty.  

We prolly need additional documentation around this now that I think about it.  A `rh-subnav` should not be used alone on a page unless otherwise `accessible-label` is set to something like `Main`.  

I guess the question is: what is the least harm, using it with multiple navigations on the page as intended and not applying a label, or having to set the label if it is the sole navigation on the page?

https://github.com/RedHat-UX/red-hat-design-system/blob/6d3d8a3e38ef18d7879826114b6f6645f6b94064/elements/rh-subnav/rh-subnav.ts#L67

File: elements/rh-subnav/rh-subnav.ts (hunk 3 above)
Comment (zeroedin) [reply elements/rh-subnav/rh-subnav.ts:3]: yup orphaned import thought I had deleted.  will clean up.

File: elements/rh-subnav/rh-subnav.ts (hunk 6 above)
Comment (adamjohnson) [reply elements/rh-subnav/rh-subnav.ts:109]: Least harm: set the label.

File: docs/assets/javascript/elements/uxdot-sidenav.ts [hunk 7]
```diff
@@ -4,6 +4,8 @@ import { classMap } from 'lit/directives/class-map.js';
 import { customElement } from 'lit/decorators/custom-element.js';
 import { property } from 'lit/decorators/property.js';
 
+import { RovingTabindexController } from '@patternfly/pfe-core/controllers/roving-tabindex-controller.js';
```
Comment (zeroedin) [inline docs/assets/javascript/elements/uxdot-sidenav.ts:7]: This change will be made against main https://github.com/RedHat-UX/red-hat-design-system/pull/1826

Comment (adamjohnson) [reply docs/assets/javascript/elements/uxdot-sidenav.ts:7]: #1826 is approved.

File: elements/rh-navigation-secondary/rh-navigation-secondary.ts (hunk 5 above)
Comment (zeroedin) [reply elements/rh-navigation-secondary/rh-navigation-secondary.ts:9]: removed a `event.preventDefault()` that was still included.  RTI would reset focus given the next index after you tabbed out of a dropdown menu, so it took control, so the preventDefault was there to ensure tab didn't first set focus.  PTAL.

File: .changeset/swift-nails-notice.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+---
+"@rhds/elements": minor
+---
+
+`<rh-navigation-secondary>`
```
Comment (adamjohnson) [inline .changeset/swift-nails-notice.md:5]:
```suggestion
`<rh-navigation-secondary>`:
```

Comment (adamjohnson) [reply .changeset/swift-nails-notice.md:5]: Also, probably worth adding a changeset for removing RTI from accordion as well.

**Review by adamjohnson — APPROVED** [review]
Minor changeset additions/changes.

File: .changeset/swift-nails-notice.md (hunk 8 above)
Comment (zeroedin) [reply .changeset/swift-nails-notice.md:5]: Oh good catch!

File: .changeset/giant-planets-argue.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+"@rhds/elements": patch
+---
+
+`<rh-accordion>`: removed Roving Tabindex keyboard navigation in favor of tab based navigation.
```
Comment (bennypowers) [inline .changeset/giant-planets-argue.md:5]:
```suggestion
`<rh-accordion>`: removed arrow-key keyboard navigation in favor of tab navigation.
```

File: .changeset/gorgeous-impalas-knock.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+---
+"@rhds/elements": minor
+---
+
+`<rh-subnav>`:
+ - removed Roving Tabindex keyboard navigation in favor of tab based navigation.
```
Comment (bennypowers) [inline .changeset/gorgeous-impalas-knock.md:6]:
```suggestion
 - removed arrow-key keyboard navigation in favor of tab navigation.
```

File: .changeset/swift-nails-notice.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
+---
+"@rhds/elements": minor
+---
+
+`<rh-navigation-secondary>`:
+ - removed Roving Tabindex keyboard navigation in favor of tab based navigation.
```
Comment (bennypowers) [inline .changeset/swift-nails-notice.md:6]:
```suggestion
 - removed arrow-key keyboard navigation in favor of tab navigation.
```

**Review by bennypowers — APPROVED** [review]
